calc/vxlan_resolver.go

```diff
@@ -263,6 +265,58 @@ func (c *VXLANResolver) OnHostConfigUpdate(update api.Update) (_ bool) {
 			})
 			c.sendVTEPRemove(nodeName)
 		}
+	case "MacVXLANTunnelAddr":
+		nodeName := update.Key.(model.HostConfigKey).Hostname
+		vtepSent := c.vtepSent(nodeName)
+		logCxt := logrus.WithField("node", nodeName).WithField("value", update.Value)
+		logCxt.Debug("MacVXLANTunnelAddr update")
+		if update.Value != nil {
+			// Update for a VXLAN tunnel MAC address.
+			newMAC := update.Value.(string)
+			currMAC := c.nodeNameToVXLANMac[nodeName]
+			logCxt = logCxt.WithFields(logrus.Fields{"newMAC": newMAC, "currMAC": currMAC})
+			if vtepSent {
+				if currMAC == newMAC {
+					// If we've already handled this node, there's nothing to do. Deduplicate.
+					logCxt.Debug("Skipping duplicate tunnel MAC addr update")
+					return
+				}
+
+				// We've already sent a VTEP for this node, and the node's MAC address has changed.
+				// We need to revoke it and any routes before sending any updates.
+				logCxt.Info("Withdrawing routes and VTEP, node changed tunnel MAC address")
+				sentSet.Iter(func(item interface{}) error {
+					r := item.(vxlanRoute)
+					if r.node == nodeName {
+						c.withdrawRoute(r)
+					}
+					return nil
+				})
+				c.sendVTEPRemove(nodeName)
+			}
+
+			// Try sending a VTEP update. If we do, this will trigger a kick of any
+			// pending routes which might now be ready to send.
+			c.nodeNameToVXLANMac[nodeName] = newMAC
+			if c.sendVTEPUpdate(nodeName) {
+				// We've successfully sent a new VTEP - check to see if any pending routes
+				// are now ready to be programmed.
+				logCxt.Info("Sent VTEP to dataplane, check pending routes")
+				c.kickPendingRoutes(pendingSet)
+			}
+		} else {
+			// Withdraw any routes which target this VTEP, followed by the VTEP itself.
```

Comment:
This is not correct.  When the MAC is removed, we should revert to the auto-generated MAC address

Reply:
Changed the code.